devices/r4isdhc.cpp

```diff
@@ -171,32 +171,94 @@ bool writeNor(const uint32_t dest_address, const uint32_t length, const uint8_t
     return true;
 }
 
-bool trySecureInit(BlowfishKey key) {
-    union {
-        uint32_t u32;
-        uint8_t bytes[4];
-    } buf;
+bool sendAndVerifyRawInit() {
+    if (ntrcard::state.status != ntrcard::Status::RAW) {
+        logMessage(LOG_ERR, "r4isdhc: sendAndVerifyRawInit: status (%d) not RAW",
+            static_cast<uint32_t>(ntrcard::state.status));
+        return false;
+    }
+
+    CmdBuf4 buf;
+    // this is actually the NOR write disable command
+    // the r4isdhc will respond to cart commands with 0xFFFFFFFF if
+    // the "magic" command hasn't been sent, so we check for that
+    sendCommand(0x40199, 4, buf.u8, 0x180000);
+    if (buf.u32 != 0xFFFFFFFF) {
+        logMessage(LOG_ERR, "r4isdhc: sendAndVerifyRawInit: pre-test returned 0x%08X", buf.u32);
+        return false;
+    }
+
+    if (platform::CAN_RESET && !ntrcard::init()) {
+        logMessage(LOG_ERR, "r4isdhc: sendAndVerifyRawInit: ntrcard::init failed");
+        return false;
+    }
+    sendCommand(0x68, 4, nullptr, 0x180000);
+    // now it will return zeroes
+    sendCommand(0x40199, 4, buf.u8, 0x180000);
+    if (buf.u32 == 0) {
+        return true;
+    }
+
+    logMessage(LOG_ERR, "r4isdhc: sendAndVerifyRawInit: post-test returned 0x%08X", buf.u32);
+    return false;
+}
+
+bool sendAndVerifyKey2Init() {
```

Comment:
I want `Key2Init` should change(yes just my opinion) to some others. it confused between `initKey2`
actually, this function only work with `old type card`.
so I suggested `~~Old` or `TypeXXX`.

it's same `sendRawInit`
and I don't understand. this is `test`, `verify` whatever like function.
why we cannot remain only this word?
because send 0x99??

nobady make printf to `setFrameBufferAndRefreshStdout`...

Reply:
`sendRawInit` sends the init-in-RAW-mode command (0x68).

`sendKey2Init` sends the init-in-KEY2-mode command (0x66).

It is **not** primarily a testing or verification function.

Reply:
it's hard to explain... why need trySecureInit.
yes they should send init commands but raw and key2 is unclear. 

1. if same naming rules, they should be `tryXXXInit`
2. you said `it's test` then `not test is verify`. I don't care actually 
3. but I guess, `xxxRawXXXX` is only can work in the new cart, and `xxxKey2XXX` is also only work in the old cart. am I misunderstood?
4. and probably nobody knows meaning until fully read code.
5. I just ask to you `these function are looks having this informations in the function name for the increase readability`.

Reply:
Key2 init works everywhere, afaik. We didn't know about the raw init command until the b9s flasher came out.